packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -116,6 +126,7 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
   }
 
   blockPeer(torrentInfoHash: string, wire: PaidStreamingWire, peerAccount: string) {
+    this.ensureEnabled();
```

Comment:
Are you sure this check should be here?
`blockPeer` and `unblockPeer` are run exclusively on a sharing peer.

So, by doing this, you are inserting checks that would disrupt the flow of messages, on a place that (I think) we really shouldn't. This would force the seeder **to not** let the leecher know that it won't respond to more requests, no matter how many funds it sends.

I think that a much better way of doing this would be that, on disabling the client, the client **blocks all it's peers, for all its torrents**, sending a BLOCK event of some sort, maybe with a special message (or with the STOP event, without the channelId as an argument that it sends right now) letting the leechers know that they shouldn't send more funds.

**Another way** is that that block gets done on a try-catch block inside the `blockPeer`, by handling the thrown error.
That way, if the user disables it's client, on `blockPeer `you could set the channelId as null or something, so that eventually when the funds of the leecher dry out, the seeder blocks new funds. And on `unblockPeer`, you just block the unblocking, because the user disabled its own funds handling capabilities.

Reply:
> blocks all it's peers, for all its torrents
this would also involve adding the handling of STOP events that don't have the `channelID `set as parameter.

Reply:
You raise a good point.

> This would force the seeder to not let the leecher know that it won't respond to more requests, no matter how many funds it sends. 

Assuming you are on master branch, if the seeder suddenly closes the browser, would the leecher end up in the same place?

> Another way is that that block gets done on a try-catch block inside the blockPeer, by handling the thrown error.

I like the second way.

Reply:
I think the situation is a bit like scenario described in https://github.com/statechannels/monorepo/issues/1215, where we need to handle peer failure (e.g., unresponsive wire, wifi going offline, browser crashing, etc.). 

We may have a "nice" peer that notifies us with a  `BLOCK` message or some sort, but we always need the logic to handle peers that purposely go offline or drop wires with malicious intent. This type of failure should probably be handled ultimately in the `ForceMove` contract / blockchain level. 

Anyway, with the discussions in https://github.com/statechannels/monorepo/issues/1215 going on, we would devise a failure handling strategy. I will take out the checks for `(un)blockPeer()` for now. The checks in `seed()` and `add()` will be still kept because these are the first calls before peers setting the wire, so it is reasonable to add there.

Reply:
I agree, this is a not so simple matter, and probably requires further discussion, to implement a wire dropped handler and stuff like that.

The seed and add checks look fine to me.